packages/haul-core-legacy/src/compiler/Fork.js

```diff
@@ -58,17 +65,15 @@ module.exports = class Fork extends EventEmitter {
       });
     }
 
-    this.isProcessing = true;
-    this.enqueuedMessages = [];
-    this.platform = platform;
+    const { port } = transportServer.address();
     this.process = createForkProcess(
-      platform,
+      this.platform,
       __dirname,
-      transportServer.options.server.address(),
-      options
+      `localhost:${port}`,
```

Comment:
is setting this to hardcoded localhost is intended?

Reply:
Yes, I have to supply some host, and the host from `transportServer` is just `::` which is invalid. This server will always run on the same machine and Haul main process. Even if the haul is run on different machine than the app or in a remote location the `localhost` should point to the same machine as Haul main process is running on.